### ISSUE TITLE
Why does it not use auto-expire set key API Jedis:setex ?

### ISSUE THREAD

**Issue by venkataramanam** [abelaska/jedis-lock#6]
I was going through the source code and found that the key is set using setnx(key, value). But that does not set the auto expiry time which has been passed to JedisLock. I was asking because the description of the lock implementation at http://redis.io/topics/distlock specifies to take an auto-release lock (with a specified expiry time).

So the Jedis equivalent of this command "SET resource_name my_random_value NX PX 30000" would probably be something like...

set(lockKey, <randomValue>, "NX", "PX", expiryTimeMillis)

Can someone explain me if my understanding is correct.

**Comment by KyleRicardo** [comment]
You are totally right, and to setNX without specify the expiry time is totolly wrong.
